packages/console-extension/src/index.ts

```diff
@@ -907,7 +908,7 @@ function activateConsoleCompleterService(
   }
 
   const trans = (translator ?? nullTranslator).load('jupyterlab');
-
+  const sanitizer = new Sanitizer();
```

Comment:
Ok we are almost there, now we should not instantiate a new sanitizer (except as fallback) but we should request the application sanitizer.

You need to add an argument `sanitizer: ISanitizer | null` to `activateConsoleCompleterService`:

https://github.com/jupyterlab/jupyterlab/blob/2a689157ebddad2fb32d0d611b349cccd10cb5b6/packages/console-extension/src/index.ts#L900-L905


```suggestion
  const sanitizer = sanitizer ?? new Sanitizer();
```

You will pass the application sanitizer by requesting its token in the plugin:

https://github.com/jupyterlab/jupyterlab/blob/2a689157ebddad2fb32d0d611b349cccd10cb5b6/packages/console-extension/src/index.ts#L211-L217

To do that, modify `optional` to be

```diff
- optional: [ICompletionProviderManager],
+ optional: [ICompletionProviderManager, ITranslator, ISanitizer],
```

> `ITranslator` needs to be added because it is missing (bug :cry:).

You need to do similar chagnes for the file editor and the notebook plugins too.

Reply:
Ah okay thanks for the review. I've added those changes now (glad we could pick up some bugs along the way 😆)